### PR TITLE
Add how to become devops engineer guide

### PR DESCRIPTION
**Pull request by dansholds**
Added new guide "How to become DevOps Engineer in 2024.

Set date as tomorrow as it's late in the day.

https://github.com/kamranahmedse/developer-roadmap/assets/25537601/919ebd05-bea0-44ef-8bae-850ddf4e5a00

### CODE REVIEW

File: src/data/guides/how-to-become-devops-engineer.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+### Infrastructure as Code (IaC)
+IaC can be seen as a superior form of automation for a DevOps engineer. IaC tools like Terraform and Amazon Cloud CDK allow the operations team to set up and manage infrastructure by simply changing a configuration file. This makes it easier to manage and scale the application.
+
+So, how do you learn these essential skills? In what sequence do you acquire these skills? The following section presents a complete roadmap on how you can learn these skills and become a DevOps engineer.
+
+
+## How to become a DevOps Engineer
+
+The internet is full of resources for learning DevOps and DevOps tools. Many of these guides lack structure and organization, resulting in a dilemma of where to start. To help beginners learn DevOps in a structured manner, [r](https://roadmap.sh)[oadmap.sh](https://roadmap.sh) has a beginner-friendly [DevOps roadmap](https://roadmap.sh/devops?r=devops-beginner).
```
Comment (kamranahmedse) [inline src/data/guides/how-to-become-devops-engineer.md:75]:
```suggestion
The internet is full of resources for learning DevOps and DevOps tools. Many of these guides lack structure and organization, resulting in a dilemma of where to start. To help beginners learn DevOps in a structured manner, [roadmap.sh](https://roadmap.sh) has a beginner-friendly [DevOps roadmap](https://roadmap.sh/devops?r=devops-beginner). 
```

File: src/data/guides/how-to-become-devops-engineer.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+**DevOps is a software engineering approach that combines development (Dev) and operations (Ops) to shorten the software development lifecycle** (feature development, bug fixes, updates) by embracing automation. It is a bridge between the development and the operations team. While the primary goal of a software engineer is to design and build software solutions, a DevOps engineer has a key role in streamlining the software delivery pipeline, automating infrastructure provisioning, and enhancing the deployment process.
+
+This article provides in-depth knowledge on how to become a DevOps engineer, highlighting the mindset and technical skills(including DevOps tools) required to become one.
+
+## TL;DR:
+DevOps combines development and operations to automate software delivery. To become a DevOps engineer, you should adopt a mindset of automation, risk awareness, systems thinking, and continuous learning. In addition to the mindset, follow the steps below:
```
Comment (kamranahmedse) [inline src/data/guides/how-to-become-devops-engineer.md:29]: I don't think it makes sense to have `h2` for TL;DR:`. I would change it to simple bold:

```suggestion
**TL;DR:** DevOps combines development and operations to automate software delivery. To become a DevOps engineer, you should adopt a mindset of automation, risk awareness, systems thinking, and continuous learning. In addition to the mindset, follow the steps below:
```

File: src/data/guides/how-to-become-devops-engineer.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+- Step 7: Continuous integration and delivery
+
+Let’s dive into the essential skills required for a [DevOps engineer](https://roadmap.sh/devops).
+
+## Skills required to become a DevOps Engineer
+
+### Operating system
+A DevOps engineer should understand the operating system, especially Linux (distribution or any variant). This is because most of the servers in use are Linux-based, so having a solid foundation in the Linux OS is imperative for infrastructure management and troubleshooting.
+
+### Solid programming fundamentals
```
Comment (kamranahmedse) [inline src/data/guides/how-to-become-devops-engineer.md:46]: Nitpick: we have a linting style to always add an empty line before and after headings

```suggestion
### Solid programming fundamentals

```

File: src/data/guides/how-to-become-devops-engineer.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+### Cloud platforms
+Procuring hardware to deploy the software is not always feasible and beneficial. Furthermore, once you purchase the hardware, you need more time and resources to manage the new hardware. Depending upon the complexity of the application, choosing cloud platforms instead of on-premise hardware can significantly reduce deployment time and cost.
+
+### Infrastructure as Code (IaC)
+IaC can be seen as a superior form of automation for a DevOps engineer. IaC tools like Terraform and Amazon Cloud CDK allow the operations team to set up and manage infrastructure by simply changing a configuration file. This makes it easier to manage and scale the application.
+
+So, how do you learn these essential skills? In what sequence do you acquire these skills? The following section presents a complete roadmap on how you can learn these skills and become a DevOps engineer.
+
+
```
Comment (kamranahmedse) [inline src/data/guides/how-to-become-devops-engineer.md:72]: One empty line at most.

```suggestion

```

File: src/data/guides/how-to-become-devops-engineer.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+### Step 6: Cloud platform
+Cloud platforms like [AWS](https://roadmap.sh/aws), Azure, and Google Cloud Platform are some popular cloud providers that allow you to deploy your projects without investing in expensive hardware. Many cloud services across these cloud platforms are analogous. So, if you are a beginner, start with AWS, as it is the most widely used cloud platform. Once you are comfortable with AWS’s services and key concepts, you can deploy the application created in earlier stages to the AWS. 
+
+Alternatively, you could demonstrate your understanding by taking the [certification exam from AWS](https://aws.amazon.com/certification/?trk=dc557659-52ab-4a28-b2b8-0b1fb90235db&sc_channel=ps&ef_id=CjwKCAjw88yxBhBWEiwA7cm6pRXCRcI-uDNNigl81tPxsbbyV9RqT6MsdSM_x5EzoFnfxcr6N22rvBoCS7IQAvD_BwE:G:s&s_kwcid=AL!4422!3!465713397254!e!!g!!aws%20certification!11138243015!111694711080&gclid=CjwKCAjw88yxBhBWEiwA7cm6pRXCRcI-uDNNigl81tPxsbbyV9RqT6MsdSM_x5EzoFnfxcr6N22rvBoCS7IQAvD_BwE).
```
Comment (kamranahmedse) [inline src/data/guides/how-to-become-devops-engineer.md:113]:
```suggestion
Alternatively, you could demonstrate your understanding by taking the [certification exam from AWS](https://aws.amazon.com/certification/).
```

**Review by kamranahmedse — CHANGES_REQUESTED** [review]

File: src/data/guides/how-to-become-devops-engineer.md (hunk 3 above)
Comment (kamranahmedse) [reply src/data/guides/how-to-become-devops-engineer.md:45]: This needs to be applied to all the headings.

Comment (dansholds) [reply src/data/guides/how-to-become-devops-engineer.md:46]: Done 👍

File: src/data/guides/how-to-become-devops-engineer.md (hunk 4 above)
Comment (dansholds) [reply src/data/guides/how-to-become-devops-engineer.md]: applied throughout
